python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -4157,8 +4151,8 @@ def test_df_astype_string_to_other(as_dtype):
     expect["foo"] = expect_data
     expect["bar"] = expect_data
 
-    got = gdf.astype(as_dtype, **kwargs)
-    assert_eq(expect, got)
+    got = gdf.dropna().astype(as_dtype, **kwargs)
+    assert_eq(expect.dropna(), got)
```

Comment:
Why do we need to dropna here?

Reply:
This was to support recent change where we error incase of `None` present in a series of type string and the typecast is to `int` types. Instead of dropping na, removed `None` from the data when typecast happens to `int32`.